src/Config/ConfigLocator.php

```diff
@@ -478,4 +485,29 @@ public function setComposerRoot($selectedComposerRoot)
     {
         $this->composerRoot = $selectedComposerRoot;
     }
+
+    /**
+     * Given an array of paths, separates files and directories.
+     *
+     * @param array $paths
+     *
+     * @return array
+     *   An array. The first row is an array of files, the second row is an
+     *   array of dirs.
+     */
+    protected function separateFilesAndDirs($paths) {
+        $files = [];
+        $dirs = [];
+        foreach ($paths as $path) {
+            if (file_exists($path)) {
+                if (is_dir($path)) {
+                    $dirs[] = realpath($path);
```

Comment:
If you pass in `$candidates` to this function then you could call `$this->identifyCandidates($dirs, $candidates);` here (or at the end of this function). This would allow you to keep `$dirs` private to this method and return only `$files`.

Reply:
This improvement would of course imply a renaming of this method.